### PR TITLE
Add test for active 3d texture bug on Mac OSX

### PR DESCRIPTION
**Pull request by qkmiao**
This will catch crash on Mac OSX 10.12 with Intel HD Graphics 515
graphics card.

### CODE REVIEW

File: sdk/tests/conformance2/textures/misc/active-3d-texture-bug.html [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+in vec2 v_coord;
+uniform mediump sampler3D u_sampler;
+out vec4 o_color;
+void main () {
+    o_color = texture(u_sampler, vec3(v_coord, 0.0));
+}
+</script>
+<script>
+"use strict";
+description("Test for a Mac OSX driver bug activing TEXTURE1 for 3d texture");
```
Comment (zhenyao) [inline sdk/tests/conformance2/textures/misc/active-3d-texture-bug.html]: activating

File: sdk/tests/conformance2/textures/misc/active-3d-texture-bug.html [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+    // Render using sampler
+    render(0, width, height, black, "Render from TEXTURE0 should be black");
+
+    gl.bindSampler(0, null);
+    gl.deleteSampler(sampler);
+
+    // Render using texture
+    render(0, width, height, black, "Render from TEXTURE0 should be black");
+    render(1, width, height, green, "Render from TEXTURE1 should be green");
```
Comment (zhenyao) [inline sdk/tests/conformance2/textures/misc/active-3d-texture-bug.html]: Maybe I misunderstood something, but why sometimes it's black and other times it's green? Whether TEXTURE0 or TEXTURE1, it's the same texture, regardless of sampling method.

File: sdk/tests/conformance2/textures/misc/active-3d-texture-bug.html [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+    gl.bindTexture(gl.TEXTURE_3D, null);
+    gl.deleteTexture(texture);
+    gl.deleteProgram(program);
+}
```
Comment (zhenyao) [inline sdk/tests/conformance2/textures/misc/active-3d-texture-bug.html]: Can you add comment either in the beginning of the code or where it crashed the detailed nature of this driver bug? Is it absolutely necessary for three passes of rendering to catch this crash?

Comment (qkmiao) [inline sdk/tests/conformance2/textures/misc/active-3d-texture-bug.html]: I added comments where driver crashed. The first two render passes are required. The last render pass isn't required to catch the crash. I just use the last render pass to show the correct drawing from TEXTURE1.

Comment (zhenyao) [inline sdk/tests/conformance2/textures/misc/active-3d-texture-bug.html]: So basically TEXTURE1 isn't necessary to reproduce the crash? Is the 3d texture also necessary?

It seems to me the relevant path to crash is:

bindSampler
render with default 3D texture + sample
deleteSampler
render with default 3D texture without sample (crash)

If that's the case, let's minimize it so it's easier for driver vendors to debug.

Comment (qkmiao) [inline sdk/tests/conformance2/textures/misc/active-3d-texture-bug.html]: Active TEXTURE1 and 3d texture are necessary to reproduce the crash.
I removed the third render pass to minimize the code.
Thanks for review. PTAL.

File: sdk/tests/conformance2/textures/misc/active-3d-texture-bug.html [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+in vec2 v_coord;
+uniform mediump sampler3D u_sampler;
+out vec4 o_color;
+void main () {
+    o_color = texture(u_sampler, vec3(v_coord, 0.0));
+}
+</script>
+<script>
+"use strict";
+description("Test for a Mac OSX driver crash bug activating TEXTURE1 for 3d texture");
```
Comment (zhenyao) [inline sdk/tests/conformance2/textures/misc/active-3d-texture-bug.html]: Sorry I didn't include this in my previous review, but can you add a comment to say this bug is on MacOSX 10.12 with Intel GPUs?

**Comment by zhenyao** [comment]
Thanks for minimizing this.  LGTM with one suggestion to add a comment.

File: sdk/tests/conformance2/textures/misc/active-3d-texture-bug.html [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+    wtu.checkCanvasRect(gl, 0, 0, width, height, expectedColor, msg);
+}
+
+function activeTextureTest() {
+    var texture = gl.createTexture();
+    var sampler = gl.createSampler();
+    var width = 64;
+    var height = 64;
+    var depth = 4;
+    var green = [0, 255, 0, 255];
```
Comment (zhenyao) [inline sdk/tests/conformance2/textures/misc/active-3d-texture-bug.html]: Also, could you remove green? It's not used now.

File: sdk/tests/conformance2/textures/misc/active-3d-texture-bug.html (hunk 4 above)
Comment (qkmiao) [inline sdk/tests/conformance2/textures/misc/active-3d-texture-bug.html]: Done

**Comment by zhenyao** [comment]
Thank you!

**Pull request merged**